mobile/app/src/main/java/com/smartjam/app/ui/screens/login/LoginViewModel.kt

```
@@ -0,0 +1,100 @@
+
+            try {
+                val result = authRepository.login(currentEmail, currentPassword)
+
+                if (result.isSuccess){
+                    eventChannel.send(LoginEvent.NavigateToHome)
+                }
+                else{
+                    val error = result.exceptionOrNull()?.message ?: "Error"
+                    _state.update { it.copy(errorMessage = error) }
+                }
+            } catch (e: Exception){
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)